crypto/provider_conf.c

```diff
@@ -236,15 +236,43 @@ static int provider_conf_load(OSSL_LIB_CTX *libctx, const char *name,
         /* First handle some special pseudo confs */
 
         /* Override provider name to use */
-        if (strcmp(confname, "identity") == 0)
+        if (strcmp(confname, "identity") == 0) {
             name = confvalue;
-        else if (strcmp(confname, "soft_load") == 0)
+        } else if (strcmp(confname, "soft_load") == 0) {
             soft = 1;
```

Comment:
Ha, this is another case where we should check for the boolean values only.

Reply:
suggest we handle this in a separate issue, or we will wind up with significant feature creep here.  Opened [a New Issue](https://github.com/openssl/openssl/issues/23105) to track it

Reply:
I am OK with this, but we absolutely must handle this new issue before 3.3.0 otherwise we will create an inconsistency here.